worldedit-core/src/main/java/com/sk89q/worldedit/internal/util/ChunkDeleter.java

```diff
@@ -0,0 +1,258 @@
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"UseOfSystemOutOrSystemErr", "CallToPrintStackTrace"})
+public class ChunkDeleter {
+    public static final Logger logger = LoggerFactory.getLogger(ChunkDeleter.class);
+
+    public static void main(String[] args) throws IOException {
+        if (args.length < 1) {
+            System.err.println("Chunk list must be specified.");
+            System.exit(1);
+            return;
+        }
+        boolean doBackup = args.length > 1 && args[1].equalsIgnoreCase("true") || promptDoBackup();
+        String chunkFile = args[0];
+
+        ChunkDeleter cd = new ChunkDeleter(doBackup);
+        if (!cd.loadChunkFile(chunkFile)) {
+            System.exit(1);
+        }
+        cd.readChunksFromFile();
+
+        System.out.print("Press [enter] to begin deletion...");
+        //noinspection ResultOfMethodCallIgnored
+        System.in.read();
+
+        cd.runDeleter();
+
+        Files.deleteIfExists(Paths.get(chunkFile));
+    }
+
+    public static void runFromFile(File delChunks, boolean backup) {
+        final ChunkDeleter chunkDeleter = new ChunkDeleter(backup);
+        if (chunkDeleter.loadChunkFile(delChunks.getPath())) {
+            try {
+                if (chunkDeleter.runDeleter()) {
+                    logger.info("Successfully deleted chunks.");
+                    if (!delChunks.delete()) {
+                        delChunks.deleteOnExit();
+                    }
+                } else {
+                    logger.error("Error occurred while deleting chunks. " +
+                            "If world errors occur, stop the server and restore the *.bak backup files.");
+                }
+            } catch (IOException e) {
+                logger.warn("Exception occurred while backup up region files. Skipping chunk deletion for safety.", e);
+            }
+        } else {
+            logger.warn("Couldn't load chunk deletion list. Skipping chunk deletion this time.");
+        }
+    }
+
+    private static boolean promptDoBackup() {
+        Scanner scanner = new Scanner(System.in);
+        Boolean backup = null;
+        System.out.print("Would you like to create backups of the region files before deleting chunks? [Y/n] ");
+        while (backup == null) {
+            String input = scanner.next();
+            if (input.isEmpty() || input.charAt(0) == 'Y' || input.charAt(0) == 'y') {
+                backup = true;
+            } else if (input.charAt(0) == 'N' || input.charAt(0) == 'n') {
+                backup = false;
+            } else {
+                System.out.print("Please enter 'y' or 'n': ");
+            }
+        }
+        return backup;
+    }
+
+    public ChunkDeleter(boolean doBackup) {
+        this.backup = doBackup;
+    }
+
+    private boolean backup;
+    private List<String> chunks;
+    private Map<String, Map<String, Set<BlockVector2>>> regionMap;
```

Comment:
Perhaps a `Table<String, String, Set<>>` would be nicer?

Reply:
I was thinking about this but I think we depend on the server classpath in all cases for guava to exist (meaning in "offline" mode we'd need to add the server jar to the sh/bat's classpath arg)